pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/invertedindex/RealtimeLuceneDocIdCollector.java

```diff
@@ -60,9 +61,16 @@ public void setScorer(Scorable scorer)
       @Override
       public void collect(int doc)
           throws IOException {
+        if (_shouldCancel) {
+          throw new RuntimeException("Lucene query was cancelled");
```

Comment:
"Cancelled" here could be misleading. 

We should be clear about timeout based interruption vs query cancelled. The latter terminology has been used for memory usage based query killing IIRC. cc @jasperjiaguo

Reply:
It looks like 'cancelled' is mostly used for cancelling streaming queries, 'killed' being used for the OOM protection

It's still inconsistent as you said so I could update it to `shouldTerminate` to keep the Collector interrupt generic and this will always be wrapped by a TimeoutException

Reply:
Per my above comment in https://github.com/apache/pinot/pull/11558#discussion_r1325017798, I am ok with the terminology aspect of cancelled for Timeout. 

However, please change the exception message to something more appropriate. See the suggestion in https://github.com/apache/pinot/pull/11558#discussion_r1325020854